src/fireflyClient.ts

```diff
@@ -237,19 +237,25 @@ export class FireflyClient {
   /**
    * initializes contract addresses & onboards user
    */
-  init = async (userOnboarding: boolean = true) => {
+  init = async (apiToken: string = "", userOnboarding: boolean = true) => {
     // get contract addresses
     const addresses = await this.getContractAddresses();
     if (!addresses.ok) {
       throw Error("Failed to fetch contract addresses");
     }
 
-    this.contractAddresses = addresses.data;
+     this.contractAddresses = addresses.data;
 
     // onboard user if not onboarded
-    if (userOnboarding) {
+    if (userOnboarding && !apiToken) {
```

Comment:
we can do something like for the sake of ease and simplicity

if(apiToken){
        //do xyz..
    }
    else if(userOnboarding){
        //do onboarding
    }